src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -167,4 +169,31 @@ public static DetailAST getAnnotation(final DetailAST ast,
         return result;
     }
 
+    /**
+     * Checks if the given AST element is marked with an annotation that allows
+     * it to omit javadoc documentation.
+     * @param ast The type or method definition.
+     * @param allowedAnnotations A collection of annotations that allow omitting documentation.
+     * @return true if the type or method is exempt from comments, false otherwise.
+     */
+    public static boolean hasAllowedAnnotations(DetailAST ast, List<String> allowedAnnotations) {
+        boolean result = false;
+        final DetailAST annotationHolderNode = getAnnotationHolder(ast);
+        DetailAST annotationNode = annotationHolderNode.findFirstToken(TokenTypes.ANNOTATION);
+        while (annotationNode != null && annotationNode.getType() == TokenTypes.ANNOTATION) {
```

Comment:
Along with the above change, this could become a `for` loop similar to https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java#L153-L167

Reply:
I moved this code from its original location, I don't think I should be making more changes to it. What is the benefit of changing a while loop into a for loop?

Reply:
@rnveach , please reply.

Reply:
I think this is obsolete because of my refactoring there is no special loop anymore